client/mksubvolume.cc

```diff
@@ -332,36 +368,20 @@ doit()
 
     // Execute all steps.
 
-    char* tmp_mountpoint = strdup("/tmp/mksubvolume-XXXXXX");
-    if (!mkdtemp(tmp_mountpoint))
-	throw runtime_error_with_errno("mkdtemp failed", errno);
-
-    do_tmp_mount(fs, tmp_mountpoint, subvol_option);
+    TmpMountpoint tmp_mountpoint("/tmp/mksubvolume-XXXXXX", fs, subvol_option);
 
     try
     {
-	do_create_subvolume(tmp_mountpoint, subvolume_name);
+	do_create_subvolume(tmp_mountpoint.mountpoint.get(), subvolume_name);
     }
-    catch (...)
+    catch (const runtime_error_with_errno& e)
     {
-	// Rethrow the original exception, not a potential exception of do_tmp_umount.
-	try
-	{
-	    do_tmp_umount(fs, tmp_mountpoint);
-	    rmdir(tmp_mountpoint);
-	    free(tmp_mountpoint);
-	}
-	catch (...)
-	{
-	}
-
-	throw;
+	if (e.error_number == EEXIST)
+	    cout << "subvolume exists already" << endl;
```

Comment:
What happens if e.g. a directory already exists where the subvolume should be created?

Reply:
Not much, as nothing has to be done here - the destructor of TmpMountpoint will be called to clean up everything, and mksubvolume will continue with creating the fstab entry.

Reply:
But adding the fstab entry for anything not a subvolume is wrong. This will lead to boot failures.

Reply:
Ah, I see: You mean when it's just a directory instead of a subvolume. At least on openSUSE I hope that will never be the case, as that would mean someone modified the '@' subvolume, but I agree this may happen more easily on other distributions...

Reply:
Detecting conflicting files in the latest commit now.